pyvista/core/celltype.py

```diff
@@ -3,18 +3,244 @@
 from __future__ import annotations
 
 from enum import IntEnum
+import textwrap
+from typing import Literal
+from typing import NamedTuple
 
 from . import _vtk_core as _vtk
 
+_DROPDOWN_TEMPLATE = """
+.. dropdown:: More info
+    :icon: info
 
-class CellType(IntEnum):
+{}
+"""
+
+_GRID_TEMPLATE_NO_IMAGE = """
+.. grid:: 1
+    :margin: 1
+
+    .. grid-item::
+
+{}{}{}
+"""
+
+_GRID_TEMPLATE_WITH_IMAGE = """
+.. grid:: 1 2 2 2
+    :reverse:
+    :margin: 1
+
+    .. grid-item::
+        :columns: 12 4 4 4
+
+        .. card::
+            :class-body: sd-px-0 sd-py-0 sd-rounded-3
+
+            .. image:: /../_build/plot_directive/api/examples/_autosummary/pyvista-examples-cells-{}-1_00_00.png
+
+    .. grid-item::
+        :columns: 12 8 8 8
+
+{}{}{}
+"""
+
+
+def _indent_paragraph(string, level):
+    indentation = "".join(['    '] * level)
+    return textwrap.indent(textwrap.dedent(string).strip(), indentation)
+
+
+# See link for color names: https://sphinx-design.readthedocs.io/en/latest/badges_buttons.html
+_BADGE_COLORS = dict(linear='primary', primary='success', dimension='secondary', geometry='muted')
+
+
+def _generate_linear_badge(is_linear: bool):
+    text = 'Linear' if is_linear else 'Non-linear'
+    return f":bdg-{_BADGE_COLORS['linear']}:`{text}`"
+
+
+def _generate_primary_badge(is_primary: bool):
+    text = 'Primary' if is_primary else 'Composite'
+    return f":bdg-{_BADGE_COLORS['primary']}:`{text}`"
+
+
+def _generate_dimension_badge(dimension: int):
+    return f":bdg-{_BADGE_COLORS['dimension']}:`{dimension}D`"
+
+
+def _generate_points_badge(num_points: int):
+    return f":bdg-{_BADGE_COLORS['geometry']}:`Points: {num_points}`"
+
+
+def _generate_edges_badge(num_edges: int):
+    return f":bdg-{_BADGE_COLORS['geometry']}:`Edges: {num_edges}`"
+
+
+def _generate_faces_badge(num_faces: int):
+    return f":bdg-{_BADGE_COLORS['geometry']}:`Faces: {num_faces}`"
+
+
+class _CellTypeTuple(NamedTuple):
+    value: int
+    cell_class: type[_vtk.vtkCell] | None = None
+    short_doc: str | None = None
+    long_doc: str | None = None
+    example: str | None = None
+    points_override: Literal['variable', 'n/a'] | None = None
+    edges_override: Literal['variable', 'n/a'] | None = None
+    faces_override: Literal['variable', 'n/a'] | None = None
+
+
+class _DocIntEnum(IntEnum):
```

Comment:
I think we can get rid of this class, it only contains `CellType` specific code and can't really be reused. Just call it `CellType`  instead